packages/react-avatar/SPEC.md

```diff
@@ -0,0 +1,272 @@
+
+## Sample Code
+
+Basic examples:
+
+```jsx
+<Avatar name="Miguel Garcia" />
+<Avatar size={72} name="Mona Kane" image="./MonaKane.jpg" />
+<Avatar square icon={<IDBadgeIcon />} />
+```
+
+Displaying a badge (\*\*subject to change pending final spec for the `Badge` component):
+
+```jsx
+<Avatar name="Allan Munger" badge={<PresenceBadge status="busy">} />
+```
+
+With active state indication:
+
+```jsx
+<Avatar name="Daisy Phillips" active={true} activeDisplay="ring-shadow" />
+<Avatar name="Robin Counts" active={false} activeDisplay="ring-shadow" />
+```
+
+## Variants
+
+### Shape
+
+The Avatar natively supports a circular and square shape. However, some use cases require a hexagon shape (e.g. to represent a bot). The hexagon will not be supported directly, but it is possible to style the Avatar with a hexagon background by using `tokens`:
+
+```jsx
+<Avatar
+  icon={<ChatBotIcon />}
+  display="icon"
+  tokens={{
+    width: 'calc(var(--avatar-height) * 1.125)',
+    background: `url('${hexagonSvg}') 0px/contain no-repeat`,
+    borderRadius: '0',
+  }}
+/>
+```
+
+Where hexagonSvg is defined as:
+
+```js
+hexagonSvg =
+  'data:image/svg+xml;utf8,' +
+  '<svg width="36" height="32" viewBox="0 0 36 32" fill="none" xmlns="http://www.w3.org/2000/svg">' +
+  '<path fill="rgb(232,232,232)" d="M0.407926 17.528C-0.135976 16.5859 -0.135975 15.4141 0.407926 14.472' +
+  'L7.91541 1.46793C8.44076 0.557947 9.39444 0 10.4245 0H25.5755C26.6056 0 27.5592 0.557951 28.0846 1.46793' +
+  'L35.5921 14.472C36.136 15.4141 36.136 16.5859 35.5921 17.528L28.0846 30.5321' +
+  'C27.5592 31.4421 26.6056 32 25.5755 32H10.4245C9.39443 32 8.44076 31.4421 7.91541 30.5321L0.407926 17.528Z"/>' +
+  '</svg>';
+```
+
+## API
+
+From [Avatar.types.tsx](https://github.com/microsoft/fluentui/blob/master/packages/react-avatar/src/components/Avatar/Avatar.types.tsx):
+
+### Slots
+
+- `image` - The Avatar's image, if available.
+- `label` - The text shown when there's no image. Defaults to the initials derived from `name` using `getInitials`.
+- `icon` - Icon displayed when there's no image or intials available.
+- `badge` - Badge to show the avatar's status.
+
+### Props
+
+```ts
+export interface AvatarProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /** The Avatar's image. */
+  image?: ShorthandValue<{}>;
+
+  /** The label shown when there's no image. Defaults to the initials derived from `name` using `getInitials`. */
+  label?: ShorthandValue<{}>;
+
+  /** Icon displayed when there's no image or intials available, or if `display="icon"`. */
+  icon?: ShorthandValue<{}>;
+
+  /** What the avatar displays. This can be used to override the default behavior, for example to show the icon even
+   *  if the avatar has initials that could be shown as the label. */
+  display?: 'image' | 'label' | 'icon';
```

Comment:
The `display` prop is not strictly necessary, but it allows overriding the default behavior of always showing the image if provided. Technically this could also be achieved by simply not setting the image or name.

One use case it does allow for that wouldn't otherwise work, is if you want to supply a `name` for accessibility purposes, but show an icon instead of initials. However, I'm not aware of anyone needing that support.

Reply:
In v0 we are using priority for this:

https://github.com/microsoft/fluentui/blob/d820d6ee1aafcd0a1669ba5bc139db8ab0663a92/packages/fluentui/react-northstar/src/components/Avatar/Avatar.tsx#L144-L145

Render if present `image` => `icon` => `label`.

---

```jsx
<Avatar display="image" name="John Doe" />
```

What it will render in this?

Reply:
Avatar has a similar priority for choosing what to display as in v0. The `display` prop would be used to re-order that priority.

In the example you gave, it would display the initials. Setting `display="image"` makes image the highest priority if present (which it already is by default), but then it falls back to the default priority if image is missing.

Reply:
> Avatar has a similar priority for choosing what to display as in v0. The `display` prop would be used to re-order that priority.

IMO in this case `display` prop should force one of these variants.
I don't see this prop in v7/v8, was this behavior requested by customers?

Reply:
No this wasn't specifically requested, but it was somewhat of a consequence of the design that had the priority as `image` => `text` => `icon` for most cases, but also `image` => `icon` (never displaying text) for others. In the latter case, it'd be impossible to specify the `name` prop (for accessibility, tooltip, etc.) but still show the icon instead of the text. It seemed like being able to be explicit about which one to show would make it more straightforward.

I'll double check with design about the priority order for the props, and whether the special case is needed.

Reply:
After talking with design, we can remove this prop and update the priority to match v0: `image` => `icon` => `label`.

The design spec'ed behavior of showing name instead of icon for people/groups can just be usage guidance: only set `icon` if there is no `name` available.